internal/storage/db/internal/db.go

```diff
@@ -464,6 +478,55 @@ func (s *dbStorage) GetDependents(ctx context.Context, ids ...namer.ModuleID) (m
 	return out, nil
 }
 
+func (s *dbStorage) Disable(ctx context.Context, policyKey ...string) (int64, error) {
+	if len(policyKey) == 1 {
```

Comment:
I think that this optimization is not necessary because in practice `disable` should be called very rarely. This just complicates the code for very little benefit.